### PR TITLE
Fix analyzer warnings

### PR DESCRIPTION
**Pull request by olmisa**
* [ ] ~Has `CHANGELOG.md` been updated?~
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] ~Did you add unit tests?~
* [ ] ~Did you check UI tests on the sample app? They are not executed on CI.~
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Static analyzer warnings were fixed.

## Related PRs or issues

[AB#80700](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/80700)

### CODE REVIEW

File: AppCenter/AppCenter/Internals/Context/UserId/MSUserIdContext.m [hunk 1]
```diff
@@ -79,7 +79,7 @@ - (NSString *)userId {
 - (void)setUserId:(nullable NSString *)userId {
   NSArray *synchronizedDelegates;
   @synchronized(self) {
-    BOOL sameUserId = (!userId && !self.currentUserIdInfo.userId) || [self.currentUserIdInfo.userId isEqualToString:(NSString *)userId];
+    BOOL sameUserId = (!userId && !self.currentUserIdInfo.userId) || (userId && self.currentUserIdInfo.userId && [self.currentUserIdInfo.userId isEqualToString:(NSString *)userId]);
```
Comment (MatkovIvan) [inline AppCenter/AppCenter/Internals/Context/UserId/MSUserIdContext.m:82]: Wrong change

File: AppCenter/AppCenter/Internals/HttpClient/MSHttpCall.m [hunk 2]
```diff
@@ -34,7 +34,9 @@ - (instancetype)initWithUrl:(NSURL *)url
     if (data && ![mutableHeaders objectForKey:kMSHeaderContentTypeKey]) {
       mutableHeaders[kMSHeaderContentTypeKey] = kMSAppCenterContentType;
     }
-    _data = data;
+    if (data) {
```
Comment (MatkovIvan) [inline AppCenter/AppCenter/Internals/HttpClient/MSHttpCall.m:37]: Why do we need it?

File: SasquatchMac/SasquatchMacObjC/AppDelegate.m [hunk 3]
```diff
@@ -41,7 +41,7 @@ - (void)applicationDidFinishLaunching:(NSNotification *)notification {
 
   // Set max storage size.
   NSNumber *storageMaxSize = [[NSUserDefaults standardUserDefaults] objectForKey:kMSStorageMaxSizeKey];
-  if (storageMaxSize) {
+  if (storageMaxSize.boolValue) {
```
Comment (MatkovIvan) [inline SasquatchMac/SasquatchMacObjC/AppDelegate.m:44]: it is not bool

Comment (olmisa) [reply SasquatchMac/SasquatchMacObjC/AppDelegate.m]: Fixed

File: AppCenter/AppCenter/Internals/HttpClient/MSHttpCall.m (hunk 2 above)
Comment (olmisa) [reply AppCenter/AppCenter/Internals/HttpClient/MSHttpCall.m]: Reworked

File: AppCenter/AppCenter/Internals/Context/UserId/MSUserIdContext.m (hunk 1 above)
Comment (olmisa) [reply AppCenter/AppCenter/Internals/Context/UserId/MSUserIdContext.m]: Reworked

File: AppCenter/AppCenter/Internals/Context/UserId/MSUserIdContext.m [hunk 4]
```diff
@@ -79,7 +79,7 @@ - (NSString *)userId {
 - (void)setUserId:(nullable NSString *)userId {
   NSArray *synchronizedDelegates;
   @synchronized(self) {
-    BOOL sameUserId = (!userId && !self.currentUserIdInfo.userId) || [self.currentUserIdInfo.userId isEqualToString:(NSString *)userId];
+    BOOL sameUserId = (!userId && !self.currentUserIdInfo.userId) || (userId && [self.currentUserIdInfo.userId isEqualToString:(NSString *)userId]);
```
Comment (annakocheshkova) [inline AppCenter/AppCenter/Internals/Context/UserId/MSUserIdContext.m:82]: Does this change the logic? Has this part been tested?

File: Sasquatch/SasquatchObjC/AppDelegate.m [hunk 5]
```diff
@@ -104,11 +104,11 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
...
-                                                                                                message:@"The maximum size of the internal "
-                                                                                                        @"storage could not be set."
+                       UIAlertController *alertController = [UIAlertController alertControllerWithTitle:NSLocalizedString(@"Warning!", nil)
+                                                                                                message:NSLocalizedString(@"The maximum size of the internal "
+                                                                                                        @"storage could not be set.", nil)
                                                                                          preferredStyle:UIAlertControllerStyleAlert];
-                       [alertController addAction:[UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault handler:nil]];
+                       [alertController addAction:[UIAlertAction actionWithTitle:NSLocalizedString(@"OK", nil) style:UIAlertActionStyleDefault handler:nil]];
```
Comment (annakocheshkova) [inline Sasquatch/SasquatchObjC/AppDelegate.m:111]: Is that right? First argument seems to be the __key__: https://developer.apple.com/documentation/foundation/nslocalizedstring

Comment (olmisa) [reply Sasquatch/SasquatchObjC/AppDelegate.m:111]: I tested these changes. Crashed the sasquatch puppet and set a storage max size as a very large number. All alerts were shown as expected.

File: AppCenter/AppCenter/Internals/Context/UserId/MSUserIdContext.m (hunk 4 above)
Comment (olmisa) [reply AppCenter/AppCenter/Internals/Context/UserId/MSUserIdContext.m:82]: As I understand, _isEqualToString_ returns _true_ only when both parameters are not _nil_ and equals each other. 
A case when both are _nil_ considered in the first part. The second part returns _true_ for the same cases as it was before (both _non-nil_ and equals each other).

File: AppCenter/AppCenter/Internals/Storage/MSDBStorage.m [hunk 6]
```diff
@@ -23,7 +23,9 @@ + (void)load {
 }
 
 - (instancetype)initWithSchema:(MSDBSchema *)schema version:(NSUInteger)version filename:(NSString *)filename {
-  _schema = schema;
+  if (schema) {
```
Comment (MatkovIvan) [inline AppCenter/AppCenter/Internals/Storage/MSDBStorage.m:26]: Same as above

File: AppCenter/AppCenter/MSAppCenter.m [hunk 7]
```diff
@@ -323,8 +323,12 @@ - (BOOL)configureWithAppSecret:(NSString *)appSecret
 }
 
 - (void)start:(NSString *)secretString withServices:(NSArray<Class> *)services fromApplication:(BOOL)fromApplication {
-  NSString *appSecret = [MSUtility appSecretFrom:secretString];
-  NSString *transmissionTargetToken = [MSUtility transmissionTargetTokenFrom:secretString];
+  NSString *appSecret;
+  NSString *transmissionTargetToken;
+  if (secretString) {
```
Comment (MatkovIvan) [inline AppCenter/AppCenter/MSAppCenter.m:328]: Can we adjust parameters nullability instead?

File: AppCenterAnalytics/AppCenterAnalytics/MSAnalytics.m [hunk 8]
```diff
@@ -450,7 +450,10 @@ - (void)resumeTransmissionTargetForToken:(NSString *)token {
 - (id<MSChannelUnitProtocol>)oneCollectorChannelUnit {
   if (!_oneCollectorChannelUnit) {
     NSString *oneCollectorGroupId = [NSString stringWithFormat:@"%@%@", self.groupId, kMSOneCollectorGroupIdSuffix];
-    self.oneCollectorChannelUnit = [self.channelGroup channelUnitForGroupId:oneCollectorGroupId];
+    id<MSChannelUnitProtocol> tempUnit = [self.channelGroup channelUnitForGroupId:oneCollectorGroupId];
+    if (tempUnit) {
```
Comment (MatkovIvan) [inline AppCenterAnalytics/AppCenterAnalytics/MSAnalytics.m:454]: Same

File: AppCenterAnalytics/AppCenterAnalytics/MSAnalytics.m [hunk 9]
```diff
@@ -459,7 +462,10 @@ - (void)resumeTransmissionTargetForToken:(NSString *)token {
   if (!_oneCollectorCriticalChannelUnit) {
     NSString *oneCollectorCriticalGroupId =
         [NSString stringWithFormat:@"%@_%@%@", self.groupId, kMSCriticalChannelSuffix, kMSOneCollectorGroupIdSuffix];
-    self.oneCollectorCriticalChannelUnit = [self.channelGroup channelUnitForGroupId:oneCollectorCriticalGroupId];
+    id<MSChannelUnitProtocol> tempUnit = [self.channelGroup channelUnitForGroupId:oneCollectorCriticalGroupId];
+    if (tempUnit) {
```
Comment (MatkovIvan) [inline AppCenterAnalytics/AppCenterAnalytics/MSAnalytics.m:466]: Same

File: AppCenter/AppCenter/Internals/Storage/MSDBStorage.m (hunk 6 above)
Comment (olmisa) [reply AppCenter/AppCenter/Internals/Storage/MSDBStorage.m]: Fixed

File: AppCenter/AppCenter/MSAppCenter.m (hunk 7 above)
Comment (olmisa) [reply AppCenter/AppCenter/MSAppCenter.m]: Fixed

File: AppCenterAnalytics/AppCenterAnalytics/MSAnalytics.m (hunk 8 above)
Comment (olmisa) [reply AppCenterAnalytics/AppCenterAnalytics/MSAnalytics.m]: Fixed

File: AppCenterAnalytics/AppCenterAnalytics/MSAnalytics.m (hunk 9 above)
Comment (olmisa) [reply AppCenterAnalytics/AppCenterAnalytics/MSAnalytics.m]: Fixed

File: AppCenter/AppCenter/Internals/Vendor/Reachability/MS_Reachability.m [hunk 10]
```diff
@@ -148,13 +148,13 @@ - (void)stopNotifier {
...
-      CFRelease(reachabilityRef);
-    }];
-  }
+    __block SCNetworkReachabilityRef reachabilityRef = self.reachabilityRef;
+    if (reachabilityRef != NULL) {
+      [MSDispatcherUtil performBlockOnMainThread:^{
+        SCNetworkReachabilityUnscheduleFromRunLoop(reachabilityRef, CFRunLoopGetCurrent(), kCFRunLoopDefaultMode);
+        CFRelease(reachabilityRef);
+      }];
+    }
```
Comment (MatkovIvan) [inline AppCenter/AppCenter/Internals/Vendor/Reachability/MS_Reachability.m:157]: This formatting change doesn't seem correct

Comment (olmisa) [reply AppCenter/AppCenter/Internals/Vendor/Reachability/MS_Reachability.m]: Fixed

**Pull request merged**